### PR TITLE
add phone verification code feature

### PR DESCRIPTION
**Pull request by Inclet**
#### What does this PR do?
This PR sends the phone verification code to a user.
#### Description of Task to be completed?
- Created a controller for sending a verification code.
-  Added test for this feature
#### How should this be manually tested?
After cloning this repo from https://github.com/Inclet/Ironji_mobile_api.git, 
checkout to the folder(Ironj_mobile_api), and run `npm install`, to start the server run `npm run dev`
send `phoneNumber` in the request body of `localhost:5000/api/user/send_phone_verification_code` with a `PUT method`
#### Any background context you want to provide?
Use postman( a chrome extension) to send request to the server.
#### Screenshots (if appropriate)
Screen Shot 2019-06-02 at 05 16 39

### CODE REVIEW

File: controllers/userController.js [hunk 1]
```diff
@@ -11,26 +21,87 @@ class User {
...
+
+        if(phoneNumber[0] !== '7'){
+            return res.status(400).send({
+                status: 400,
+                error: 'Invalid Phone Number.'
+            })
+        }
+
+        nexmo.verify.request({
+            number: `250${phoneNumber}`,
```
Comment (ogeniuspriority) [inline controllers/userController.js]: so you have chosen to make all phones start with 250. I think allowing a number to go in as it is is better kubera number iva kuri front end ifite signature ya +...country code, plus number so it's already okay with the number from the phone. ntwara +2500000000000, or another country code.

File: controllers/userController.js [hunk 2]
```diff
@@ -11,26 +21,87 @@ class User {
...
+                status: 400,
+                error: 'Phone Number Should Contain Numbers Only.'
+            })
+        }
+
+        if(phoneNumber[0] === '0'){
+           phoneNumber = phoneNumber.substr(1);
+        }
+
+        if(phoneNumber.length > 9) {
```
Comment (ogeniuspriority) [inline controllers/userController.js]: Number of digits may vary according to country unless ushaka ko dukoresha 250 gusa

File: controllers/userController.js (hunk 1 above)
Comment (Inclet) [reply controllers/userController.js]: yeah, nasanze nshatse kuyigira scalable byasaba ko nkoresha ibintu byinshi kuko each country has its own numbering system. like mexico is only 8 digit and other countries are less than seven or even above 10 digits.

File: controllers/userController.js (hunk 2 above)
Comment (Inclet) [reply controllers/userController.js]: Just because it's an MVP we need to start by using local only.

**Pull request merged**